### PR TITLE
Fix error when deserializing datatypes from number

### PR DESCRIPTION
**Pull request by Aegonek**
https://github.com/time-rs/time/issues/505
Please see my comment in the issue.

### CODE REVIEW

File: src/serde/visitor.rs [hunk 1]
```diff
@@ -204,7 +204,23 @@ impl<'a> de::Visitor<'a> for Visitor<Weekday> {
...
             7 => Ok(Weekday::Sunday),
             _ => Err(E::invalid_value(
-                de::Unexpected::Unsigned(value.into()),
+                de::Unexpected::Signed(value),
+                &"a value in the range 1..=7",
+            )),
+        }
+    }
+
+    fn visit_u64<E: de::Error>(self, value: u64) -> Result<Weekday, E> {
```
Comment (jhpratt) [inline src/serde/visitor.rs:213]: Both `visit_i64` and `visit_u64` are required to capture all integer types, from what I can tell? Excluding 128 bit types, but that's a separate story.

File: tests/integration/serde/json.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    assert_eq!(serialize(March, Readable)?, "\"March\"");
+    assert_eq!(deserialize::<Month>("\"March\"", Readable)?, March);
+    assert_eq!(deserialize::<Month>("3", Readable)?, March);
+
+    Ok(())
+}
+
+#[test]
+fn time_json() -> Result<(), Box<dyn Error>> {
+    let time = Time::from_hms(12, 40, 20)?;
```
Comment (jhpratt) [inline tests/integration/serde/json.rs:77]: For statically known values, it's best to use the macro.

**Review by jhpratt — CHANGES_REQUESTED** [review]
Couple changes. CI should also pass — right now there seems to be quite a bit uncovered. [Here's a link](https://app.codecov.io/gh/time-rs/time/pull/506) to the coverage for this PR.

File: src/serde/visitor.rs (hunk 1 above)
Comment (Aegonek) [reply src/serde/visitor.rs:213]: Yes, it seems to be the case. In this commit I added `visit_u64` implementation for every `visit_i64` implementation.

File: tests/integration/serde/json.rs (hunk 2 above)
Comment (Aegonek) [reply tests/integration/serde/json.rs]: :+1:

**Comment by Aegonek** [comment]
> Couple changes. CI should also pass — right now there seems to be quite a bit uncovered. [Here's a link](https://app.codecov.io/gh/time-rs/time/pull/506) to the coverage for this PR.

Yeah, Codecov says that `visit_i64` is not tested. It seems that `serde_json` will always delegate `deserialize_u8` to `visit_u64` for our use case. It may delegate it to `deserialize_i64` if the number is negative and `deserialize_f64` if the number is greater than `u64::MAX` - but we know that won't happen. I think the best course of action is ^1: to remove `visit_i64` implementation.

Alternative (^2) - add implementation for `visit_f64`, opt out from code coverage for `visit_i64` and `visit_f64`? Given that those methods exist mainly for "better safe than sorry", I think that would also be fine. Although keeping methods you don't expect to be used by framework is a bit smelly, so I prefer option ^1.

**Review by jhpratt — APPROVED** [review]
I just pushed up a small diff to simplify some of the code, which reduces its size slightly. LGTM at this point. Thank you!

**Pull request merged**